### PR TITLE
Enable running Pull Requests against v2-10-stable branch

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
cc: @eladkal @utkarsharma2 -> you will also have to merge and backport-that one to #41610 in order to make the CI PR workflow to run

**Comment by potiuk** [comment]
We have not been doing it before because we were directly cherry-picking and only "push" events happened for v2-10-test, not PRs - by enabling branch protection, and only allowing PRs to v2* branches, we need to enable this as well

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ on:  # yamllint disable-line rule:truthy
   push:
     branches: ['v[0-9]+-[0-9]+-test']
   pull_request:
-    branches: ['main', 'v[0-9]+-[0-9]+-test']
+    branches: ['main', 'v[0-9]+-[0-9]+-test', 'v[0-9]+-[0-9]+-stable']
```
Comment (hussein-awala) [inline .github/workflows/ci.yml:26]: All stable tags or just 2.10?

**Review by hussein-awala — APPROVED** [review]
Nit

File: .github/workflows/ci.yml (hunk 1 above)
Comment (potiuk) [reply .github/workflows/ci.yml:26]: All. We don't want to make special exceptions.

**Pull request merged**